middleware/configuration/unittest/source/test_model.cpp

```diff
@@ -454,6 +454,237 @@ namespace casual
 
       }
 
+      TEST( configuration_model_remove, remove_servers_B_from_A)
+      {
+         common::unittest::Trace trace;
+
+         constexpr auto configuration = R"(
+domain:
+   name: model
+   servers:
+      - alias: a
+        path: a
+      - alias: b
+        path: b
+      - alias: c
+        path: c
+)";
+
+         constexpr auto remove = R"(
+domain:
+   name: model
+   servers:
+      - alias: b
+        path: b
+)";
+
+         constexpr auto expected = R"(
+domain:
+   name: model
+   servers:
+      - alias: a
+        path: a
+      - alias: c
+        path: c
+)";
+
+         auto A = local::configuration( configuration);
+         auto B = local::configuration( remove);
+         auto result = local::configuration( expected);
+
+         set_difference( A, std::move( B));
```

Comment:
I'm not sure if I like that set_xxx are mutating... Feels wrong to update A in this case... To me, it would be more _functional_ to have `A = set_difference( std::move( A), std::move( B));` Or, if one need to keep originals: `auto result = set:difference( A, B);`